### PR TITLE
feat: refresh ITO when expire

### PR DESCRIPTION
**Pull request by zhouhanseng**

### CODE REVIEW

File: packages/maskbook/src/plugins/ITO/UI/ITO.tsx [hunk 1]
```diff
@@ -308,6 +308,18 @@ export function ITO(props: ITO_Props) {
...
     )
 
+    useEffect(() => {
+        const timeToExpired = end_time * 1000 - new Date().getTime()
+        if (timeToExpired < 0 || listOfStatus.includes(ITO_Status.expired)) return
+
+        const timer = setTimeout(() => {
+            setOpenClaimDialog(false)
+            retryITOCard()
+        }, timeToExpired + 30000)
```
Comment (zhouhanseng) [inline packages/maskbook/src/plugins/ITO/UI/ITO.tsx:318]: There's 10+ seconds delay after arrives the end_time.

**Review by Jack-Works — DISMISSED** [review]

File: packages/maskbook/src/plugins/ITO/UI/ITO.tsx [hunk 2]
```diff
@@ -308,6 +308,18 @@ export function ITO(props: ITO_Props) {
...
     )
 
+    useEffect(() => {
+        const timeToExpired = end_time * 1000 - new Date().getTime()
+        if (timeToExpired < 0 || listOfStatus.includes(ITO_Status.expired)) return
+
+        const timer = setTimeout(() => {
+            setOpenClaimDialog(false)
+            retryITOCard()
+        }, timeToExpired + TIME_WAIT_BLOCKCHAIN)
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/ITO/UI/ITO.tsx:318]: Does this duration represent the average blocking time of the Ethereum chain?

**Comment by guanbinrui** [comment]
I think we should both add an expired validator in the `useClaimCallback` hook.

File: packages/maskbook/src/plugins/ITO/UI/ITO.tsx (hunk 2 above)
Comment (zhouhanseng) [reply packages/maskbook/src/plugins/ITO/UI/ITO.tsx:318]: No, it is a conservative time to ensure expired.

**Review by Jack-Works — APPROVED** [review]

**Pull request merged**